web/src/components/storage/PartitionPage.tsx

```diff
@@ -512,56 +523,269 @@ type AutoSizeProps = {
+
+      return sprintf(
+        // TRANSLATORS: %s is a mount point (eg. /)
+        _(
+          "The size range for %s will be dynamically adjusted based on the usage of Btrfs snapshots.",
+        ),
+        path,
+      );
+    }
+
+    if (otherPaths.length === 1) {
+      return sprintf(
+        // TRANSLATORS: %1$s is a mount point (eg. /) and %2$s is another one (eg. /home)
+        _(
+          "The size range for %1$s will be dynamically adjusted based on the presence of a separate file system for %2$s.",
+        ),
+        path,
+        otherPaths[0],
+      );
+    }
+
+    return sprintf(
+      // TRANSLATORS: %1$s is a mount point and %2$s is a list of other paths
+      _(
+        "The size range for %1$s will be dynamically adjusted based on the presence of separate file systems for %2$s.",
+      ),
+      path,
+      formatList(otherPaths),
+    );
+  };
+
+  const limitsText = (size) => {
+    if (size.max) {
+      if (size.max === size.min) {
+        return sprintf(
+          // TRANSLATORS: %s is a size with units (eg. 10 GiB)
+          _("The current configuration will result in a partition of %s."),
+          deviceSize(size.min),
+        );
+      }
+
+      return sprintf(
+        // TRANSLATORS: %1$s is a min size, %2$s is the max size
+        _(
+          "The current configuration will result in a partition with a size between %1$s and %2$s.",
+        ),
+        deviceSize(size.min),
+        deviceSize(size.max),
+      );
+    }
+
+    return sprintf(
+      // TRANSLATORS: %s is a size with units (eg. 10 GiB)
+      _("The current configuration will result in a partition of at least %s."),
+      deviceSize(size.min),
+    );
+  };
+
+  const intro = introText(volume);
+  const limits = limitsText(size);
+  // TRANSLATORS: used just to concatenate the explanation about how a size is calculated and the
+  // range resulting of that calculation.
+  return sprintf(_("%1$s %2$s"), intro, limits);
+}
+
+function autoSizeText(volume, size): string {
+  const path = volume.mountPath;
+
+  if (path) {
+    if (volume.autoSize) {
+      const otherPaths = volume.outline.sizeRelevantVolumes || [];
+
+      if (otherPaths.length || volume.outline.snapshotsAffectSizes) {
+        return autoSizeTextDynamic(volume, size);
+      }
+
+      // This assumes volume.autoSize is correctly set. Ie. if it is set to true then at least one
+      // of the relevant outline fields (snapshots, RAM and sizeRelevantVolumes) is used.
+      return autoSizeTextRam(path, size);
+    }
+
+    return autoSizeTextFixed(path, size);
   }
 
-  // return <Alert variant="info" isInline isPlain title={conditionsText} />;
-  return <Content component="blockquote">{conditionsText}</Content>;
+  // Fallback volume
+  // This assumes the fallback volume never uses automatic sizes (ie. re-calculated based on
+  // other aspects of the configuration). It would be VERY surprising if that's the case.
+  return autoSizeTextFallback(size);
```

Comment:
If you agree with my comment above about using the same texts for the fallback volume, then we could remove the `if (path)` condition and avoid the assumption you mention there in the comment.

Reply:
I don't. ;-)